nautobot/extras/models/change_logging.py

```diff
@@ -28,10 +28,35 @@ class ChangeLoggedModel(models.Model):
 
     created = models.DateTimeField(auto_now_add=True, blank=True, null=True)
     last_updated = models.DateTimeField(auto_now=True, blank=True, null=True)
+    created_by = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
+    last_updated_by = models.ForeignKey(
+        to=settings.AUTH_USER_MODEL,
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        related_name="+",
+    )
 
     class Meta:
         abstract = True
 
+    def save(self, *args, **kwargs):
+        from nautobot.extras.choices import ObjectChangeActionChoices
+
+        # call save() first so that the ObjectChange record exists
+        super().save(*args, **kwargs)
+        most_recent_change = get_changes_for_model(self).first()
+        if most_recent_change and most_recent_change.action == ObjectChangeActionChoices.ACTION_CREATE:
```

Comment:
From Glenn: Would it make any sense to instead do this as part of the change_logging decorator, instead of the object save()? Just thinking about bulk operations and avoiding the need to repeatedly look up the ObjectChange and associated User for each and every object created/modified.

Reply:
Why do we need to the `ObjectChange` record to exist first before attempting to change the `created_by` or `last_updated_by`?

Reply:
Because as currently implemented we're using the ObjectChange to find the user that needs to be set into those fields as we don't natively have that information in `save()`.